app/hotels/src/allHotels/searchForm/locationPicker/LocationPickerScreen.js

```diff
@@ -0,0 +1,185 @@
+
+import * as React from 'react';
+import { View } from 'react-native';
+import {
+  Text,
+  Touchable,
+  StyleSheet,
+  Color,
+  TextInput,
+} from '@kiwicom/react-native-app-shared';
+import { type NavigationType } from '@kiwicom/react-native-app-navigation';
+import { PublicApiRenderer } from '@kiwicom/react-native-app-relay';
+import { graphql } from 'react-relay';
+
+import type { LocationPickerScreen_cities_QueryResponse as LocationSuggestions } from './__generated__/LocationPickerScreen_cities_Query.graphql';
+import SuggestionList from './SuggestionList';
+
+type Props = {|
+  navigation: NavigationType,
+|};
+
+type NavigationProps = {|
+  cityId: string,
+  cityName: string,
+  navigation: NavigationType,
+|};
+
+type State = {|
+  search: string,
+  cityId: string,
+|};
+
+const styles = StyleSheet.create({
+  cancelButton: {
+    padding: 8,
+  },
+  headerButtonText: {
+    fontSize: 17,
+    color: Color.white,
+    lineHeight: 20,
+    fontFamily: 'SFProText-Regular',
+  },
+  confirmButtonText: {
+    fontFamily: 'SFProText-Semibold',
+  },
+  headerTitle: {
+    color: Color.white,
+    fontFamily: 'SFProText-Semibold',
+    fontSize: 17,
+    lineHeight: 20,
+  },
+  confirmButton: {
+    padding: 8,
+    paddingRight: 10,
+  },
+  disabled: {
+    opacity: 0.4,
+  },
+  container: {
+    flex: 1,
+  },
+  textInputContainer: {
+    backgroundColor: Color.white,
+    padding: 10,
+  },
+  input: {
+    borderRadius: 2,
+    borderWidth: 1,
+    borderColor: '#c0c8d1',
+    paddingLeft: 11,
+    paddingBottom: 10,
+    paddingTop: 13,
+  },
+});
+
+export default class LocationPicker extends React.Component<Props, State> {
+  state = {
+    search: '',
+    cityId: '',
+  };
+
+  static navigationOptions = ({
+    navigation,
+    cityId,
+    cityName,
+  }: NavigationProps) => {
+    function goBack() {
+      navigation.goBack();
+    }
+
+    function confirmClicked() {
+      // Using navigation.navigate({ params }) actually renders the screen inside the modal
+      // Solution is go back and pass an on select function from parent
+      // https://github.com/react-navigation/react-navigation/issues/288#issuecomment-315684617
+      goBack();
+      navigation.state.params.onSelect(cityId, cityName);
+    }
+
+    function headerLeft() {
+      return (
+        <Touchable
+          borderlessRipple
+          onPress={goBack}
+          style={styles.cancelButton}
+        >
+          <Text style={[styles.headerButtonText, styles.confirmButtonText]}>
+            Cancel
+          </Text>
+        </Touchable>
+      );
+    }
+
+    function headerRight() {
+      return (
+        <Touchable
+          onPress={confirmClicked}
+          style={styles.confirmButton}
+          borderlessRipple
+        >
+          <Text style={styles.headerButtonText}>Confirm</Text>
+        </Touchable>
+      );
+    }
+
+    return {
+      headerLeft: headerLeft(),
+      title: <Text style={styles.headerTitle}>Where</Text>,
```

Comment:
Why do you use the `Text` component here? Our default setting looks alright...

Reply:
It is a custom font, default is SFUI, this header has SFProText-Semibold,

Reply:
Seems like a design bug to me - why to complicate it? I would prefer to have one title font everywhere...

Reply:
Ok, I will leave as the others